src/pykeen/nn/modules.py

```diff
@@ -2549,54 +2558,182 @@ def forward(
+        base_kwargs: OptionalKwargs = None,
+        clamp_score: Clamp | float | None = None,
+    ):
         """
         Initialize the interaction module.
 
+        :param base:
+            the base interaction.
+        :param base_kwargs:
+            keyword-based parameters used to instantiate the base interaction
         :param clamp_score:
             whether to clamp scores into a fixed interval
         """
         super().__init__()
         if isinstance(clamp_score, float):
             clamp_score = (-clamp_score, clamp_score)
         self.clamp_score = clamp_score
+        self.base = interaction_resolver.make(base, base_kwargs)
 
-    # docstr-coverage: inherited
-    def _prepare_state_for_functional(self) -> MutableMapping[str, Any]:  # noqa: D102
-        return dict(clamp=self.clamp_score)
+    @property
+    def entity_shape(self) -> Sequence[str]:  # type:ignore[override]
+        """Expose the base interaction's entity shape."""
+        return self.base.entity_shape
+
+    @property
+    def relation_shape(self) -> Sequence[str]:  # type:ignore[override]
+        """Expose the base interaction's relation shape."""
+        return self.base.relation_shape
 
     # docstr-coverage: inherited
-    @staticmethod
-    def _prepare_hrt_for_functional(
-        h: HeadRepresentation,
-        r: RelationRepresentation,
-        t: TailRepresentation,
-    ) -> MutableMapping[str, FloatTensor]:  # noqa: D102
-        return dict(h=h[0], h_inv=h[1], r=r[0], r_inv=r[1], t=t[0], t_inv=t[1])
+    def forward(self, h: HeadRepresentation, r: RelationRepresentation, t: TailRepresentation) -> FloatTensor:
+        scores = self.base(h, r, t)
+        if self.clamp_score is None:
+            return scores
+        low, high = self.clamp_score
+        return torch.clamp(scores, min=low, max=high)
+
+
+class DirectionAverageInteraction(
+    Interaction[
+        tuple[FloatTensor, FloatTensor],
+        tuple[FloatTensor, FloatTensor],
+        tuple[FloatTensor, FloatTensor],
+    ],
+):
+    r"""The directional average interaction module.
+
+    This can be considered as a generalization of the SimplE interaction module that can be parametrized
+    with any other interaction module, rather than just DistMult.
+
+    A separate representation is learned for each entity $e \in \mathcal{E}$  for when it appears as the
+    subject of a triple $\mathbf{h} \in \mathbb{R}^d$ and as the object of a triple
+    $\mathbf{t} \in \mathbb{R}^d$.
+
+    Two representations are learned for each relationship for a forward $\textbf{r}_{\rightarrow}$
+    and backward triple $\textbf{r}_{\leftarrow}$.
+
+    Depending on whether an entity participates in a
+    triple as the head or tail entity, either $\mathbf{e}_h$ or $\mathbf{e}_t$ is used. Both entity
+    representations are learned independently, i.e. observing a triple $(h,r,t)$, the method only updates
+    $\mathbf{h}_h$ and $\mathbf{t}_t$.
```

Comment:
I think this part is not true, but rather a shortcoming of e.g., `CPInteraction` that motivated `SimplE`; In SimplE, we use the head entity's tail-representation for the reverse direction.

Reply:
okay, I was going to ask about this since I thought I misunderstood. Maybe you can take a pass at re-writing this

Reply:
done in https://github.com/pykeen/pykeen/pull/1470/commits/99fbd4ef6cd8a6e0859c951c3a399949fa2927bd